### ISSUE TITLE
require-jsdoc not to throw 'Missing JSDoc comment' error when function has no parameters and any returns

### ISSUE THREAD

**Issue by tohjg** [eslint/eslint#10761]
**Please describe what the rule should do:**
I'm not sure if this has being implemented but I can't find any documentation or could it be any plugin? 

Is there any configuration not to throw `Missing JSDoc comment` error by require-jsdoc when __function has no arguments and no returns__.

**What category of rule is this? (place an "X" next to just one item)**

[X] Enforces code style
[X] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// Do not warn "Missing JSDoc comment"
function meaningfulDescribedFunction() {
  ...
}

// Warn "Missing JSDoc comment" when there is parameter
function anotherFunctionWithParam(foo) {
  ...
}

// Warn "Missing JSDoc comment" when there is returns
function anotherFunctionWithReturnsVar() {
  const foo = 'bar';
  return foo
}

// Warn "Missing JSDoc comment" when there is parameter and returns
function anotherFunctionWithReturnsVar(baz) {
  ...
  const foo = 'bar';
  return foo
}

```

**Why should this rule be included in ESLint (instead of a plugin)?**
require-jsdoc is part of eslint. I checked on jsdoc plugin and can't find any. I [raised the same question to them](https://github.com/gajus/eslint-plugin-jsdoc/issues/77). Function naming can be something meaningful and descriptive, thus making just a function description is merely a duplication. Too many comments make code harder to read.

**Comment by tohjg** [comment]
@kaicataldo 

Sorry for delay. 

I'm not sure what's different "report an error" and "runtime error". But I guess runtime error should an error occurred inside the linting tools itself.

What I intended to suggest is that it should report "Missing JSDoc comment" to tell developer to create doc block if function has parameters and/or returns during linting process (in VSCode's Problems tab or terminal, eg `yarn js:lint`). So I guess this is "report an error".

Not sure if we are in the same boat. If is there anything unclear, please let me know. 🙂

**Comment by nzakas** [comment]
Thanks for the suggestion. As of yesterday, we have decided to officially [end-of-life JSDoc support](https://eslint.org/blog/2018/11/jsdoc-end-of-life) in ESLint. All JSDoc features are now deprecated and we won't be fixing bugs or making any further improvements to those features.

We are recommending that people transition over to use the [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc) plugin instead of the built-in rules in ESLint.

Thanks for understanding and we apologize for any inconvenience.
